### PR TITLE
Indexdb

### LINKED ISSUE #31: Spaces and underscores unpredictable on mac

### LINKED ISSUE #32: IndexDB implementation
**Briefly describe the required task**
- Local Storage has 5mb limit

**Briefly describe how the solution might be implemented**
- Use IndexedDB

### PR DESCRIPTION
**Pull request by dangeo314**
## Workload Includes
1. Changed all local storage aspects to indexedDB.

## Tests Done
1. Manual testing of upload, delete, download, edit/save.

## Potential Improvements
1. More testing

### CODE REVIEW

File: source/assets/scripts/database.mjs [hunk 1]
```diff
@@ -0,0 +1,121 @@
+const DB_NAME = 'imagedb';
+const DB_VERSION = 2; // Use a long long for this value (don't use a float)
+const DB_STORE_NAME = 'images';
+
+var db;
+export async function openDb() {
```
Comment (nkarter02) [inline source/assets/scripts/database.mjs:6]: going to need js doc commenting

File: source/assets/scripts/database.mjs [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+export async function deleteImage(img_name){
+    return new Promise( function(resolve, reject) {
+        const tx = db.transaction(DB_STORE_NAME, 'readwrite');
+        const imageStore = tx.objectStore(DB_STORE_NAME); 
+        const request = imageStore.delete(img_name);
+        request.onerror = (event) => { //name found in db
+            console.error("error deleting " + img_name + " from db")
+            reject("error");
+        };
+        request.onsuccess = (event) => {
```
Comment (nkarter02) [inline source/assets/scripts/database.mjs:57]: yeah definitely going to need comments since idk whats going on

**Review by nkarter02 — CHANGES_REQUESTED** [review]
pretty much just needs comments, variable naming and code styling is good

**Review by mchera24 — COMMENTED** [review]
Looks good

**Review by nkarter02 — APPROVED** [review]

**Pull request merged**